### PR TITLE
Fix MSIX packaging to determine if a Preview release by inspecting the semantic version string

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

The non-Preview release still gets a Product Name of "PowerShell Preview" because the code was looking to see if the `$productname` contains a `-`.  Since the version is added after the product name, it will always have a `-` such as `powershell-7.0.0`.  The fix is to only look for the `-` within the semantic version string like `7.0.0-rc.1`.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -3049,7 +3049,7 @@ function New-MSIXPackage
```
Comment (PoshChan) [inline tools/packaging/packaging.psm1]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/ADOPTERS.md.https://shell.azure.com/ should work
```powershell
Failed to complete request to "https://shell.azure.com/". Response status code does not indicate success: 500 (Internal Server Error).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
119:                                     throw "Failed to complete request to `"$url`". $($_.Exception.Message)"

```

Comment (PoshChan) [inline tools/packaging/packaging.psm1]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.0.30', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

**Comment by adityapatwardhan** [comment]
@PoshChan please retry static, macos

**Comment by PoshChan** [comment]
@adityapatwardhan, successfully started retry of `PowerShell-CI-static-analysis, PowerShell-CI-macOS`

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (PoshChan) [inline tools/packaging/packaging.psm1]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/ADOPTERS.md.https://shell.azure.com/ should work
```powershell
Failed to complete request to "https://shell.azure.com/". Response status code does not indicate success: 500 (Internal Server Error).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
119:                                     throw "Failed to complete request to `"$url`". $($_.Exception.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/ADOPTERS.md.https://shell.azure.com/ should work
```powershell
Failed to complete request to "https://shell.azure.com/". Response status code does not indicate success: 500 (Internal Server Error).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
119:                                     throw "Failed to complete request to `"$url`". $($_.Exception.Message)"

```

**Comment by SteveL-MSFT** [comment]
@PoshChan retry static

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-static-analysis`

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (PoshChan) [inline tools/packaging/packaging.psm1]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/ADOPTERS.md.https://shell.azure.com/ should work
```powershell
Failed to complete request to "https://shell.azure.com/". Response status code does not indicate success: 500 (Internal Server Error).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
119:                                     throw "Failed to complete request to `"$url`". $($_.Exception.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/ADOPTERS.md.https://shell.azure.com/ should work
```powershell
Failed to complete request to "https://shell.azure.com/". Response status code does not indicate success: 403 (Site Disabled).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
119:                                     throw "Failed to complete request to `"$url`". $($_.Exception.Message)"

```

**Comment by SteveL-MSFT** [comment]
@PoshChan retry static

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-static-analysis`

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**